k-distribution/src/main/scripts/bin/kx

```diff
@@ -0,0 +1,442 @@
+fi
+
+kompiledDir=
+hasKompiledDir=false
+for file in "$dir"/*-kompiled; do
+  kompiledDir=$file
+  if $hasKompiledDir; then
+    error 'Multiple compiled definitions found in the current working directory: ' "$dir"/*-kompiled
+  fi
+  hasKompiledDir=true
+done
+
+if ! $hasKompiledDir; then
+  error 'Could not find a compiled definition. Use --directory to specify one.'
+fi
+
+hasArgv=false
+if [[ "${#ARGV[@]}" -gt 0 ]]; then
+  config_var_PGM="${ARGV[0]}"
+  hasArgv=true
+  params+=("PGM")
+fi
+
+mainModuleName=$(cat "$kompiledDir"/mainModule.txt)
+
+if [ "$outputFile" = "-" ]; then
+  outputFile=/dev/stdout
+fi
+
+if $term; then
+  if [ -z "${parser_PGM+unset}" ]; then
+    execute kast -m "$mainModuleName" "$config_var_PGM" -o kore > "$input_file"
+  else
+    execute $parser_PGM "$config_var_PGM" > "$input_file"
+  fi
+else
+  for name in "${params[@]}"; do
+    parser_name="parser_$name"
+    config_name="config_var_$name"
+    tempFile="$(mktemp tmp.in."$name".XXXXXXXXXX)"
+    tempFiles+=("$tempFile")
+    sortName=`cat "$kompiledDir"/sort_"$name".txt`
+    if [ -z "${!parser_name+unset}" ]; then
+      if [ -f "$kompiledDir/parser_$name" ]; then
+        parser="$kompiledDir/parser_$name"
+      elif [ "$name" = "PGM" ]; then
+        if $hasArgv; then
+          parser="kast -o kore"
+	else
+          parser="kast -m $mainModuleName -o kore"
+	fi
+      else
+        parser="kast -m $mainModuleName -o kore"
+      fi
+    else
+      parser="${!parser_name}"
+    fi
+    execute $parser "${!config_name}" > "$tempFile"
+    configVars="$configVars -c $name $tempFile $sortName korefile"
+  done
+  if [ "$(cat "$kompiledDir"/sort_IO.txt 2>/dev/null)" = "String" ]; then
+    if $io; then
+      ioText=on
+    else
+      ioText=off
+    fi
+    configVars="$configVars -c IO \dv{SortString{}}(\"$ioText\") String kore"
+  fi
+  if [ "$(cat "$kompiledDir"/sort_STDIN.txt 2>/dev/null)" = "String" ]; then
+    if $io; then
+      stdinText=
+    else
+      unescaped=$(</dev/stdin)
+      stdinText=$(awk 'BEGIN {for(n=0;n<256;n++)ord[sprintf("%c",n)]=n} {text = text $0 ORS} END {split(text, chars, ""); for (i=1; i<=length(text); i++) { if (chars[i] == "\"") { printf("%s", "\\\"") } else if (chars[i] == "\\") { printf("%s", "\\\\") } else if (chars[i] == "\n") { printf("%s", "\\n") } else if (chars[i] == "\t") { printf("%s", "\\t") } else if (chars[i] == "\r") { printf("%s", "\\r") } else if (chars[i] == "\f") { printf("%s", "\\f") } else if (ord[chars[i]] >= 32 && ord[chars[i]] < 127) { printf("%s", chars[i]) } else { printf("\\x%02x", ord[chars[i]]) } } }' <<< "$unescaped")
+    fi
+    stdinFile="$(mktemp tmp.stdin.XXXXXXXXXX)"
+    tempFiles+=("$stdinFile")
+    printf '\dv{SortString{}}(\"%s\")\n' "$stdinText" > "$stdinFile"
+    configVars="$configVars -c STDIN $stdinFile String korefile"
+  fi
+  execute llvm-krun $configVars -d "$kompiledDir" $flags --dry-run -o "$input_file"
```

Comment:
It looks like `flags` is only intended to be for the llvm backend, maybe it should be named `llvm_flags` instead?

Reply:
Well, this code is going to be used by the haskell backend code path as well to construct the initial configuration, so I'm not sure that makes sense.